src/lib/address/address.ts

```diff
@@ -8,14 +9,45 @@ import {
   validateUnstoppableDomain,
 } from 'lib/address/unstoppable-domains'
 import { resolveYat, validateYat } from 'lib/address/yat'
+import { logger } from 'lib/logger'
 import { store } from 'state/store'
 
 import { ensReverseLookupShim } from './ens'
 
+const moduleLogger = logger.child({ namespace: ['lib', 'address'] })
+
 type VanityAddressValidatorsByChainId = {
   [k: ChainId]: ValidateVanityAddress[]
 }
 
+type ParseUrlByChainId = (chainId: ChainId, url: string) => ParseUrlByChainIdReturn | undefined
+
+type ParseUrlByChainIdReturn = {
+  target_address: string
+  parameters?: {
+    value: string
+    gas: string
+    gasPrice: string
+  }
+}
+
+// @TODO: Implement BIP21
+const parseUrlByChainId: ParseUrlByChainId = (chainId: ChainId, url: string) => {
```

Comment:
gh pr checkout 2789

Reply:
also can we please make this have an arity of 1 and only accept an object of type `ParseAddressInputArgs`, as per all the other functions in this abstraction